processing/src/main/java/org/apache/carbondata/processing/newflow/row/CarbonRowBatch.java

```diff
@@ -17,27 +17,40 @@
 
 package org.apache.carbondata.processing.newflow.row;
 
-import java.util.ArrayList;
 import java.util.Iterator;
-import java.util.List;
 
 /**
  * Batch of rows.
  */
-public class CarbonRowBatch {
+public class CarbonRowBatch implements Iterator<CarbonRow> {
 
-  private List<CarbonRow> rowBatch = new ArrayList<>();
+  private CarbonRow[] rowBatch;
```

Comment:
The default capacity of ArrayList is 10.  Before adding a element, need to ensure capacity, and grow capacity if needed.